lib/dma.h

```diff
@@ -164,15 +164,44 @@ bit_to_u8off(size_t val)
     return val % (CHAR_BIT);
 }
 
+static inline void
+u8_set_all(uint8_t *bm)
+{
+    *bm = ~0;
+}
+
+static inline void
+mask_any_lower_bits(uint8_t *byte, size_t start_idx)
+{
+    if (bit_to_u8off(start_idx) != 0) {
+        *byte &= ~((1 << bit_to_u8off(start_idx)) - 1);
+    }
+}
+
+static inline void
+mask_any_higher_bits(uint8_t *byte, size_t end_idx)
+{
+    if (bit_to_u8off(end_idx) != 0) {
+        *byte &= ((1 << bit_to_u8off(end_idx)) - 1);
+    }
+}
+
+static inline void
+apply_byte_mask(uint8_t *bitmap, uint8_t byte_mask)
```

Comment:
it's not a mask if we're or'ing. Can we just call this u8_atomic_or() to match other naming?

Reply:
ACK